tests/test_particleContainer.py

```diff
@@ -374,6 +376,7 @@ class Config:
     for lvl in range(pc.finest_level + 1):
         # loop local tiles of particles
         for pti in pc.iterator(pc, level=lvl):
+            print(f"pti.soa().real_names={pti.soa().real_names}")
```

Comment:
This loop should be identical to https://github.com/AMReX-Codes/amrex/blob/25.01/Tests/Particles/NamedSoAComponents/main.cpp#L106-L111 but somehow we loose the name info right now.

Reply:
I wonder if an iterator copy is happening and it forgets to copy `ParIterBase_impl::m_pc` somewhere.

Reply:
fixed.